src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,81 @@
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../types/Comment';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onAdd: (val: Omit<Comment, 'id'>) => Promise<void>;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onAdd, selectedPost }) => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [name, setName] = useState('');
+  const [isErrorName, setIsErrorName] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [isErrorEmail, setIsErrorEmail] = useState(false);
+
+  const [comment, setComment] = useState('');
+  const [isErrorComment, setIsErrorComment] = useState(false);
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setIsErrorName(false);
+  };
+
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    setIsErrorEmail(false);
+  };
+
+  const handleCommentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setComment(e.target.value);
+    setIsErrorComment(false);
+  };
+
+  const handleReset = () => {
+    setComment('');
+    setEmail('');
+    setName('');
+    setIsErrorComment(false);
+    setIsErrorEmail(false);
+    setIsErrorName(false);
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+
+    setIsErrorName(!name.trim());
+    setIsErrorEmail(!email.trim());
+    setIsErrorComment(!comment.trim());
+
+    if (!name.trim() || !email.trim() || !comment.trim()) {
+      return;
+    }
+
+    setIsSubmitting(true);
+    onAdd({
+      name: name.trim(),
+      email: email.trim(),
+      body: comment.trim(),
+      postId: selectedPost?.id || 0,
+    })
+      .catch(error => {
```

Comment:
You should handle errors in a more user-friendly way. Just throwing an error will break the user interface. It's better to show some message to the user. Also, it's a good practice to log the error for debugging purposes.